assets/src/web-stories-block/block/components/storiesPreview.js

```diff
@@ -36,10 +43,16 @@ function StoriesPreview(props) {
     attributes: { align, viewType, sizeOfCircles, fieldState, numOfColumns },
     viewAllLabel,
     stories,
+    instanceId,
   } = props;
 
   const { archiveURL } = useConfig();
 
+  const carouselContainer = useRef(null);
+  const carouselNext = useRef(null);
+  const carouselPrev = useRef(null);
+  const carouselItem = useRef(null);
```

Comment:
So this is one ref that is used for all items. I guess this works because they all have the same dimensions that you can check for in the `useEffect` below?

Reply:
right, so on instance of web stories will have all the items of same dimensions. So this tells the glider how much space is needed for single item and glider calculates the rest.